zipline/pipeline/filters/filter.py

```diff
@@ -427,3 +428,29 @@ class Latest(LatestMixin, CustomFilter):
     Filter producing the most recently-known value of `inputs[0]` on each day.
     """
     pass
+
+
+class SingleAsset(Filter):
+    """
+    A Filter that computes to True only for the given asset.
+    """
+    inputs = []
+    window_length = 1
+
+    def __new__(cls, asset):
+        return super(SingleAsset, cls).__new__(cls, asset=asset)
+
+    def _init(self, asset, *args, **kwargs):
+        self._asset = asset
+        return super(SingleAsset, self)._init(*args, **kwargs)
+
+    @classmethod
+    def static_identity(cls, asset, *args, **kwargs):
+        return (
+            super(SingleAsset, cls).static_identity(*args, **kwargs), asset,
+        )
+
+    def _compute(self, arrays, dates, assets, mask):
+        asset_column = searchsorted(list(assets), self._asset.sid)
+        mask[:, asset_column] = True
+        return mask
```

Comment:
Yes, though I think this will still fail if the requested asset isn't in the assets array (say, because the asset doesn't exist during the requested timeframe.)